compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1131,31 +1133,13 @@ setRootConfig(func::FuncOp entryPointFn,
 
   // Choose the next non-zero tile size immediately after the distribution
   // level to help compute the distribution tile sizes.
-  SmallVector<int64_t> distTileSizes;
-  auto vecPreProcStrategy = getVectorPreProcStrategy(linalgOp);
-  bool usePeelingPipeline =
-      vecPreProcStrategy == VectorPreProcStrategy::Peeling;
-
-  LLVM_DEBUG(KD_DBGS() << "Vector pre-processing strategy: "
-                       << vecPreProcStrategy << "\n");
-
-  if (usePeelingPipeline && isX86(targetAttr)) {
-    // It's inspired from https://github.com/iree-org/iree-llvm-sandbox repo.
-    // Sandbox has [[288, 128, 512], [12, 32, 1]] setup. We scale 288 to 192
-    // because 288/12*8=192
-    if (numLoops == 3) {
-      distConfig.maxTileSizes[0] = 192;
-      distConfig.maxTileSizes[1] = 128;
-    }
+  for (auto [cacheTileSize, vecTileSize] :
+       llvm::zip_equal(cacheTileSizes, vecTileSizes)) {
+    int64_t minTileSize = cacheTileSize != 0 ? cacheTileSize : vecTileSize;
+    distConfig.minTileSizes.push_back(minTileSize);
```

Comment:
I believe that this part was accidentally removed by one of my commits when fixing conflicts. (I'm sorry about that.) I just put it back here. @dcaballe let me know if you want me to create a separate PR to revamp this.

Reply:
Oh, yes, please. Otherwise my distribution work won't work :)